### PR TITLE
Nouran6 gamal arctan2 numpy frontend

### PR DESCRIPTION
**Pull request by Nouran6Gamal**

### CODE REVIEW

**Review by Nouran6Gamal — COMMENTED** [review]
need to review changes

**Comment by sherry30** [comment]
Hi,

the test seems to be failing for `arctan2`, can you look into it.
Let me know if you need any help with it, I'd be happy to answer any question.

Thanks

**Comment by Nouran6Gamal** [comment]
Yes please, I don't know what can I do for pass the checks?

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_trigonometric_functions.py [hunk 1]
```diff
@@ -311,3 +311,54 @@ def test_numpy_cosh(
...
+    dtype,
+    where,
+    as_variable,
+    with_out,
+    num_positional_args,
+    native_array,
+    fw,
+):
+    input_dtype, x = dtype_and_x
+    input_dtype = [input_dtype]
```
Comment (sherry30) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_trigonometric_functions.py:339]: remove this line

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_trigonometric_functions.py [hunk 2]
```diff
@@ -311,3 +311,54 @@ def test_numpy_cosh(
         subok=True,
         test_values=False,
     )
+
+    
+# arctan2
+@given(
```
Comment (sherry30) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_trigonometric_functions.py:317]: add `@handle_cmd_line_args` above this

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

Requested changes on specific lines.
Also can you look into the lint errors caused by your changes,

Thanks

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_trigonometric_functions.py (hunk 2 above)
Comment (Nouran6Gamal) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_trigonometric_functions.py:318]: done

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_trigonometric_functions.py (hunk 1 above)
Comment (Nouran6Gamal) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_trigonometric_functions.py:340]: which one 339 ?

Comment (Nouran6Gamal) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_trigonometric_functions.py]: really  I tried multiple times , through my trial I only passed one test (and the passed test varies)and error in lint decreased to 3 or 4 errors then after multiple trials the errors increased again and I can't find where is the problem , any thing else I can fix ?

Comment (Nouran6Gamal) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_trigonometric_functions.py]: finally one test passed , lint errors descreased. but I'm afraid if I change code , lint error 'll increase again 

lint error are limited to those
ivy/functional/frontends/numpy/mathematical_functions/trigonometric_functions.py:1[8](https://github.com/unifyai/ivy/runs/8159720972?check_suite_focus=true#step:5:9)5:23: E231 missing whitespace after ','
ivy/functional/frontends/numpy/mathematical_functions/trigonometric_functions.py:1[9](https://github.com/unifyai/ivy/runs/8159720972?check_suite_focus=true#step:5:10)3:1: W391 blank line at end of file
ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_trigonometric_functions.py:319:84: E231 missing whitespace after ','
ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_trigonometric_functions.py:319:89: E501 line too long (98 > 88 characters)

Comment (Nouran6Gamal) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_trigonometric_functions.py]: ivy/stateful/initializers.py:11 in public class `Initializer`:
        D204: 1 blank line required after class docstring (found 0) 
that's the only visible error in docstring test , how can I handle it

**Comment by Nouran6Gamal** [comment]
now only lint test passed
ivy/stateful/initializers.py:11 in public class `Initializer`:
        D204: 1 blank line required after class docstring (found 0) 

=========================== short test summary info ============================
FAILED ivy/ivy_tests/test_docstrings.py::test_docstrings[jax] - AssertionError: 
FAILED ivy/ivy_tests/test_docstrings.py::test_docstrings[numpy] - AssertionEr...
FAILED ivy/ivy_tests/test_docstrings.py::test_docstrings[tensorflow] - Assert...
FAILED ivy/ivy_tests/test_docstrings.py::test_docstrings[torch] - AssertionEr...
======================= 4 failed, 124 warnings in 33.98s =======================  


that's the only visible error in all errors ( in docstring test) , how can I handle both
@sherry30 @CatB1t

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

you shouldn't make changes to anything other than `arctan2`, can you remove all the extra changes.
You don't have to worry about any other test failing like in docstring etc. Just the `arctan2` test should pass in test-frontend-numpy.

thanks

**Review by Nouran6Gamal — COMMENTED** [review]
any changes in initializer.py I have already undone them , If  trying multiple times 'd cause error I can fork the repo again and edit again 
test-frontend-numpy / run-nightly-tests (pull_request)  outputs this error without describing any thing what should I do 
Run exit 1
Error: Process completed with exit code 1.

**Comment by sherry30** [comment]
You can see the errors under `Run Frontend Tests` tab, the errors don't show up under `Check on Faliures` tab

**Comment by Nouran6Gamal** [comment]
ivy/ivy_tests/test_ivy/helpers.py:3250: in new_fn
2022-09-05T08:57:09.5872162Z     return test_fn(*args, **kwargs)
2022-09-05T08:57:09.5872387Z ivy/ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_trigonometric_functions.py:318: in test_numpy_arctan2
2022-09-05T08:57:09.5872455Z     @given(
2022-09-05T08:57:09.5872677Z ivy/ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_trigonometric_functions.py:348: in test_numpy_arctan2
2022-09-05T08:57:09.5872797Z     np_frontend_helpers.test_frontend_function(
2022-09-05T08:57:09.5872960Z ivy/ivy_tests/test_ivy/test_frontends/test_numpy/helpers.py:49: in test_frontend_function
2022-09-05T08:57:09.5873095Z     _test_frontend_function_ignoring_unitialized(*args, **kwargs)
2022-09-05T08:57:09.5873292Z ivy/ivy_tests/test_ivy/test_frontends/test_numpy/helpers.py:39: in _test_frontend_function_ignoring_unitialized
2022-09-05T08:57:09.5873451Z     helpers.value_test(ret_np_flat=ret_flat, ret_np_from_gt_flat=frontend_ret_flat)
2022-09-05T08:57:09.5873544Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2022-09-05T08:57:09.5873549Z 
2022-09-05T08:57:09.5873621Z     def value_test(
2022-09-05T08:57:09.5873687Z         *,
2022-09-05T08:57:09.5873760Z         ret_np_flat,
2022-09-05T08:57:09.5873841Z         ret_np_from_gt_flat,
2022-09-05T08:57:09.5873911Z         rtol=None,
2022-09-05T08:57:09.5874018Z         atol=1e-6,
2022-09-05T08:57:09.5874116Z         ground_truth_backend="TensorFlow",
2022-09-05T08:57:09.5874180Z     ):
2022-09-05T08:57:09.5874323Z         """Performs a value test for matching the arrays in ret_np_flat and
2022-09-05T08:57:09.5874402Z         ret_from_np_flat.
2022-09-05T08:57:09.5874464Z     
2022-09-05T08:57:09.5874536Z         Parameters
2022-09-05T08:57:09.5874628Z         ----------
2022-09-05T08:57:09.5874750Z         ret_np_flat
2022-09-05T08:57:09.5874889Z             A list (flattened) containing Numpy arrays. Return from the
2022-09-05T08:57:09.5874975Z             framework to test.
2022-09-05T08:57:09.5875052Z         ret_from_np_flat
2022-09-05T08:57:09.5875197Z             A list (flattened) containing Numpy arrays. Return from the ground
2022-09-05T08:57:09.5875279Z             truth framework.
2022-09-05T08:57:09.5875334Z         rtol
2022-09-05T08:57:09.5875426Z             Relative Tolerance Value.
2022-09-05T08:57:09.5875494Z         atol
2022-09-05T08:57:09.5875585Z             Absolute Tolerance Value.
2022-09-05T08:57:09.5875666Z         ground_truth_backend
2022-09-05T08:57:09.5875769Z             Ground Truth Backend Framework.
2022-09-05T08:57:09.5875834Z     
2022-09-05T08:57:09.5875890Z         Returns
2022-09-05T08:57:09.5875992Z         -------
2022-09-05T08:57:09.5876126Z         None if the value test passes, else marks the test as failed.
2022-09-05T08:57:09.5876196Z         """
2022-09-05T08:57:09.5876351Z         if type(ret_np_flat) != list:
2022-09-05T08:57:09.5876445Z             ret_np_flat = [ret_np_flat]
2022-09-05T08:57:09.5876549Z         if type(ret_np_from_gt_flat) != list:
2022-09-05T08:57:09.5876650Z             ret_np_from_gt_flat = [ret_np_from_gt_flat]
2022-09-05T08:57:09.5876777Z >       assert len(ret_np_flat) == len(ret_np_from_gt_flat), (
2022-09-05T08:57:09.5876898Z             "len(ret_np_flat) != len(ret_np_from_gt_flat):\n\n"
2022-09-05T08:57:09.5877035Z             "ret_np_flat:\n\n{}\n\nret_np_from_gt_flat:\n\n{}".format(
2022-09-05T08:57:09.5877138Z                 ret_np_flat, ret_np_from_gt_flat
2022-09-05T08:57:09.5877207Z             )
2022-09-05T08:57:09.5877272Z         )
2022-09-05T08:57:09.5877399Z E       AssertionError: len(ret_np_flat) != len(ret_np_from_gt_flat):
2022-09-05T08:57:09.5877469Z E       
2022-09-05T08:57:09.5877544Z E       ret_np_flat:
2022-09-05T08:57:09.5877612Z E       
2022-09-05T08:57:09.5877678Z E       []
2022-09-05T08:57:09.5877748Z E       
2022-09-05T08:57:09.5877832Z E       ret_np_from_gt_flat:
2022-09-05T08:57:09.5877884Z E       
2022-09-05T08:57:09.5877973Z E       [array(0., dtype=float16)]

**Comment by Nouran6Gamal** [comment]
ivy/ivy_tests/test_ivy/helpers.py:3250: in new_fn
2022-09-05T08:57:09.5872162Z     return test_fn(*args, **kwargs)
2022-09-05T08:57:09.5872387Z ivy/ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_trigonometric_functions.py:318: in test_numpy_arctan2
2022-09-05T08:57:09.5872455Z     @given(
2022-09-05T08:57:09.5872677Z ivy/ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_trigonometric_functions.py:348: in test_numpy_arctan2
2022-09-05T08:57:09.5872797Z     np_frontend_helpers.test_frontend_function(
2022-09-05T08:57:09.5872960Z ivy/ivy_tests/test_ivy/test_frontends/test_numpy/helpers.py:49: in test_frontend_function
2022-09-05T08:57:09.5873095Z     _test_frontend_function_ignoring_unitialized(*args, **kwargs)
2022-09-05T08:57:09.5873292Z ivy/ivy_tests/test_ivy/test_frontends/test_numpy/helpers.py:39: in _test_frontend_function_ignoring_unitialized
2022-09-05T08:57:09.5873451Z     helpers.value_test(ret_np_flat=ret_flat, ret_np_from_gt_flat=frontend_ret_flat)
2022-09-05T08:57:09.5873544Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2022-09-05T08:57:09.5873549Z 
2022-09-05T08:57:09.5873621Z     def value_test(
2022-09-05T08:57:09.5873687Z         *,
2022-09-05T08:57:09.5873760Z         ret_np_flat,
2022-09-05T08:57:09.5873841Z         ret_np_from_gt_flat,
2022-09-05T08:57:09.5873911Z         rtol=None,
2022-09-05T08:57:09.5874018Z         atol=1e-6,
2022-09-05T08:57:09.5874116Z         ground_truth_backend="TensorFlow",
2022-09-05T08:57:09.5874180Z     ):
2022-09-05T08:57:09.5874323Z         """Performs a value test for matching the arrays in ret_np_flat and
2022-09-05T08:57:09.5874402Z         ret_from_np_flat.
2022-09-05T08:57:09.5874464Z     
2022-09-05T08:57:09.5874536Z         Parameters
2022-09-05T08:57:09.5874628Z         ----------
2022-09-05T08:57:09.5874750Z         ret_np_flat
2022-09-05T08:57:09.5874889Z             A list (flattened) containing Numpy arrays. Return from the
2022-09-05T08:57:09.5874975Z             framework to test.
2022-09-05T08:57:09.5875052Z         ret_from_np_flat
2022-09-05T08:57:09.5875197Z             A list (flattened) containing Numpy arrays. Return from the ground
2022-09-05T08:57:09.5875279Z             truth framework.
2022-09-05T08:57:09.5875334Z         rtol
2022-09-05T08:57:09.5875426Z             Relative Tolerance Value.
2022-09-05T08:57:09.5875494Z         atol
2022-09-05T08:57:09.5875585Z             Absolute Tolerance Value.
2022-09-05T08:57:09.5875666Z         ground_truth_backend
2022-09-05T08:57:09.5875769Z             Ground Truth Backend Framework.
2022-09-05T08:57:09.5875834Z     
2022-09-05T08:57:09.5875890Z         Returns
2022-09-05T08:57:09.5875992Z         -------
2022-09-05T08:57:09.5876126Z         None if the value test passes, else marks the test as failed.
2022-09-05T08:57:09.5876196Z         """
2022-09-05T08:57:09.5876351Z         if type(ret_np_flat) != list:
2022-09-05T08:57:09.5876445Z             ret_np_flat = [ret_np_flat]
2022-09-05T08:57:09.5876549Z         if type(ret_np_from_gt_flat) != list:
2022-09-05T08:57:09.5876650Z             ret_np_from_gt_flat = [ret_np_from_gt_flat]
2022-09-05T08:57:09.5876777Z >       assert len(ret_np_flat) == len(ret_np_from_gt_flat), (
2022-09-05T08:57:09.5876898Z             "len(ret_np_flat) != len(ret_np_from_gt_flat):\n\n"
2022-09-05T08:57:09.5877035Z             "ret_np_flat:\n\n{}\n\nret_np_from_gt_flat:\n\n{}".format(
2022-09-05T08:57:09.5877138Z                 ret_np_flat, ret_np_from_gt_flat
2022-09-05T08:57:09.5877207Z             )
2022-09-05T08:57:09.5877272Z         )
2022-09-05T08:57:09.5877399Z E       AssertionError: len(ret_np_flat) != len(ret_np_from_gt_flat):
2022-09-05T08:57:09.5877469Z E       
2022-09-05T08:57:09.5877544Z E       ret_np_flat:
2022-09-05T08:57:09.5877612Z E       
2022-09-05T08:57:09.5877678Z E       []
2022-09-05T08:57:09.5877748Z E       
2022-09-05T08:57:09.5877832Z E       ret_np_from_gt_flat:
2022-09-05T08:57:09.5877884Z E       
2022-09-05T08:57:09.5877973Z E       [array(0., dtype=float16)]
2022-09-05T08:57:09.5877979Z 
2022-09-05T08:57:09.5878103Z ivy/ivy_tests/test_ivy/helpers.py:759: AssertionError
2022-09-05T08:57:09.5878356Z ----------------------------- Captured stderr call -----------------------------
2022-09-05T08:57:09.5878697Z WARNING:root:NumPy does not support autograd, declaring a 'variable' is identical to declaring an 'array' when using numpy backend.
2022-09-05T08:57:09.5879031Z WARNING:root:NumPy does not support autograd, declaring a 'variable' is identical to declaring an 'array' when using numpy backend.
2022-09-05T08:57:09.5879276Z ------------------------------ Captured log call -------------------------------
2022-09-05T08:57:09.5879637Z WARNING  root:gradients.py:9 NumPy does not support autograd, declaring a 'variable' is identical to declaring an 'array' when using numpy backend.
2022-09-05T08:57:09.5879990Z WARNING  root:gradients.py:9 NumPy does not support autograd, declaring a 'variable' is identical to declaring an 'array' when using numpy backend.
2022-09-05T08:57:09.5880234Z ---------------------------------- Hypothesis ----------------------------------
2022-09-05T08:57:09.5880341Z Falsifying example: test_numpy_arctan2(
2022-09-05T08:57:09.5880498Z     dtype_and_x=(['float16', 'float16'],
2022-09-05T08:57:09.5880653Z      [-0.9090909090909091, -0.9090909090909091]),
2022-09-05T08:57:09.5880768Z     dtype='float16',
2022-09-05T08:57:09.5880841Z     where=False,
2022-09-05T08:57:09.5880933Z     as_variable=[False, False],
2022-09-05T08:57:09.5880997Z     with_out=False,
2022-09-05T08:57:09.5881082Z     num_positional_args=2,
2022-09-05T08:57:09.5881174Z     native_array=[False, False],
2022-09-05T08:57:09.5881277Z     fw='numpy',
2022-09-05T08:57:09.5881343Z )
2022-09-05T08:57:09.5881433Z Falsifying example: new_fn(
2022-09-05T08:57:09.5881496Z     data=data(...),
2022-09-05T08:57:09.5881727Z     get_command_line_flags={'as_variable': None,
2022-09-05T08:57:09.5881855Z      'native_array': None,
2022-09-05T08:57:09.5881970Z      'with_out': None,
2022-09-05T08:57:09.5882087Z      'container': None,
2022-09-05T08:57:09.5882215Z      'instance_method': None,
2022-09-05T08:57:09.5882343Z      'test_gradients': None},
2022-09-05T08:57:09.5882439Z     device='cpu',
2022-09-05T08:57:09.5882725Z     f=<module 'ivy.functional.backends.numpy' from '/ivy/ivy/functional/backends/numpy/__init__.py'>,
2022-09-05T08:57:09.5882828Z     fw='numpy',
2022-09-05T08:57:09.5882892Z )
2022-09-05T08:57:09.5882995Z Falsifying example: test_numpy_arctan2(
2022-09-05T08:57:09.5883149Z     dtype_and_x=(['float16', 'float16'],
2022-09-05T08:57:09.5883303Z      [-0.9090909090909091, -0.9090909090909091]),
2022-09-05T08:57:09.5883403Z     dtype='float16',
2022-09-05T08:57:09.5883474Z     where=False,
2022-09-05T08:57:09.5883561Z     as_variable=[False, False],
2022-09-05T08:57:09.5883633Z     with_out=False,
2022-09-05T08:57:09.5883763Z     num_positional_args=2,
2022-09-05T08:57:09.5883854Z     native_array=[False, False],
2022-09-05T08:57:09.5883955Z     fw='numpy',
2022-09-05T08:57:09.5884007Z )
2022-09-05T08:57:09.5884292Z ____ test_numpy_deg2rad[cpu-ivy.functional.backends.numpy-True-False-numpy] ____
2022-09-05T08:57:09.5884298Z 
2022-09-05T08:57:09.5884595Z get_command_line_flags = {'as_variable': None, 'container': None, 'instance_method': None, 'native_array': None, ...}
2022-09-05T08:57:09.5884698Z device = 'cpu'
2022-09-05T08:57:09.5884982Z f = <module 'ivy.functional.backends.numpy' from '/ivy/ivy/functional/backends/numpy/__init__.py'>
2022-09-05T08:57:09.5885123Z fw = 'numpy', args = (), kwargs = {}
2022-09-05T08:57:09.5885128Z 
2022-09-05T08:57:09.5885208Z     @given(data=st.data())
2022-09-05T08:57:09.5885284Z >   @settings(max_examples=1)
2022-09-05T08:57:09.5885300Z 
2022-09-05T08:57:09.5885393Z ivy/ivy_tests/test_ivy/helpers.py:3222:

**Comment by Nouran6Gamal** [comment]
> You can see the errors under `Run Frontend Tests` tab, the errors don't show up under `Check on Faliures` tab
2022-09-05T08:57:09.5877399Z E AssertionError: len(ret_np_flat) != len(ret_np_from_gt_flat):
2022-09-05T08:57:09.5877469Z E
2022-09-05T08:57:09.5877544Z E ret_np_flat:
2022-09-05T08:57:09.5877612Z E
2022-09-05T08:57:09.5877678Z E []
2022-09-05T08:57:09.5877748Z E
2022-09-05T08:57:09.5877832Z E ret_np_from_gt_flat:
2022-09-05T08:57:09.5877884Z E
2022-09-05T08:57:09.5877973Z E [array(0., dtype=float16)]
that's the problem , what should I edit

**Pull request closed without merging**